app/services/claims/claim_state_transition_validator.rb

```diff
@@ -0,0 +1,51 @@
+module Claims
+  class ClaimStateTransitionValidator
```

Comment:
Just a 💭 , not a **blocker**: It would be really nice to start using [ActiveModel::Validator](http://api.rubyonrails.org/classes/ActiveModel/Validator.html) (specially for new ones), rather than implementing our own, more so because we don't need to re-implement any of the existent default checks like:

```ruby
validates_presence_of :state

# instead of 
add_error('You must select a status') if state.blank?
```

Same thing applies to the end result:

```ruby
record.valid?

# instead of
result = validator.call
result.eql?(:ok)
```

Reply:
This was a familiarising refactor and i had thought of completely moving things to a claim state transition validator but did not go down this route here (limited nature of of this PR), but is worth a spike, because i believe this will cause issues with validation of data that may become erroneous for historical claims by the point of claim processing by the case worker.